plugin/cog/cog-commons/src/main/java/it/geosolutions/imageioimpl/plugins/cog/AbstractRangeReader.java

```diff
@@ -33,7 +34,7 @@
 
     protected BasicAuthURI authUri;
     protected URI uri;
-    protected Map<Long, byte[]> data = new HashMap<>();
+    protected SoftValueHashMap<Long, byte[]> data = new SoftValueHashMap<>(10);
```

Comment:
A soft value map with 10 hard references. What if the request is "full file" and the range merge manages to get everything in a single http range request, then the full file is kept in memory with that one hard reference?

Reply:
removed the hard references count